tests/test_sync.py

```diff
@@ -92,56 +87,54 @@ def test_local_equals_API(self, pir_to_xy_response, downsample_ref):
         assert y_pred == pytest.approx(y, abs=1e-2)
 
 
-class TestDownloadParallelDataset:
-    @pytest.mark.parametrize("include_expression", [True, False])
-    @pytest.mark.parametrize("downsample_ref", [25, 50])
-    @pytest.mark.parametrize("axis", ["coronal", "sagittal"])
-    def test_patched(self, include_expression, downsample_ref, axis, monkeypatch):
-        """Does not requires internet, everything is patched.
-
-        The only thing that is unpatched is the `get_transform_simple`.
-        """
+class TestDatasetDownloader:
+    @responses.activate
+    def test_patched(self, data_folder, custom_cache_dir):
+        """Does not requires internet, everything is patched."""
 
         # Parameters
-        dataset_id = 12345
-
-        # Mocking
-        get_2d_bulk_fake = Mock(
-            return_value={
-                11111: (np.ones((2, 3)), 20),
-                22222: (np.ones((2, 3)), 50),
-            }
+        dataset_id = 123
+        include_expression = False
+        downsample_ref = 25
+        axis = "sagittal"
+
+        # Mocking the responses to request
+        with open(data_folder / "sync" / "SectionDataSet.txt") as f:
+            body_dataset = f.read()
+
+        responses.add(
+            responses.GET,
+            "https://api.brain-map.org/api/v2/data/query.json?"
+            "criteria=model::SectionDataSet,rma::criteria,[id$eq123],"
+            "rma::include,alignment3d,rma::options[start_row$eq0][num_rows$eq25000]",
+            body=body_dataset,
```

Comment:
I guess another option here would be to patch the `rma_all` function and not worry about what is happening inside of it.

The current approach kind of assumes that we see both inside of `DatasetDownloader` and `rma_all` code. Which means that if we make changes either inside of `rma_all` or `DatasetDownloder` then this test will fail.

I would lean towards just patching `rma_all` and thus this test would only fail if we introduce changes inside of `DatasetDownloader`. Also, if we only patch `rma_all` then we just need to return the minimal metadata necessary to make the `DatasetDownloader` work. Anyway, I can totally help with this since it would require some work + refactoring.

Reply:
Patched `rma_all` in cae39ef.